### PR TITLE
Ignore duplate file matches

### PR DESCRIPTION
**Pull request by falk-werner**
This pull request introduced the new option "strong", which allows to ignore duplicate file matches.

Example:
````
#logrotate.conf

# a specific log can be marked "strong"
/var/log/specific.log {
  size 10k
  rotate 3
  strong
}

# matches any other log
/var/log/*.log {
  size 100k
  rotate 3
}
````

Note that order is significant: only the first occurrence of a unique file can be marked as `strong`. 

This feature is in response to following issues:

- [Allow "overwriting" of entries](https://github.com/logrotate/logrotate/issues/436)
- [allow several matches for one log file, most specific wins](https://github.com/logrotate/logrotate/issues/38)

It is not an ideal solution for these issues, but I believe it solves the basic use cases:

- an admin can include a common directory and provide some generic matchers afterwards (as shown above)
- multiple matches can be specified for one log file _(although not the most specific wins, but the first one)_

When run in verbose mode, every duplicate issues a debug message. If `strong` is not used or it is not specified for the first occurrence, an error is issued for each duplicate (same behavior as before).

### CODE REVIEW

**Comment by otheus** [comment]
I love the commit. But why not have an option that is indicative of what is actually going on? 
``--strong`` means pretty much anything and nothing here. A better option name would be ``--first-filematch-wins`` or (a bit ambiguous) ``--ignore-duplicate-file-matches``

**Comment by falk-werner** [comment]
Indeed, finding names is the tricky part here and you are totally right, `strong` can mean anything. One early impression was `ingore-duplicates`. Would that be sufficiant? _(`ignore-duplicate-file-matches` describes perfectly what is done, but it's a bit long compared to the existing options.)_

**Comment by otheus** [comment]
You realize this is a 6+ year old problem, you're solving! :) I want to award you with a trophy or something. Anyway, In issue #38 and related, they discuss an option, such as "ignore-duplicates". I used to always hate long options. It irked me in the early 90s when GNU utils started with their `--ridiculously-long-options-that-were-short-stories`. But as I've aged (or maybe as I've become stoopider), I've seen the wisdom in this.  Which "duplicates" does that option refer to? Maybe it refers to the _target_ of the file rotation? So if you have `logfile.log` and `logfile-2022-11-17.log`, and `dateext` is enabled, do you overwrite? Or ignore _that_ duplicate? Maybe someday in the future, some guy will come along and add an option for that case and be confused. I dunno. 

Also, take a look at my comment at the end of issue #38 . Consider if this change might break the implementations they *want* there. Yeah, I know it's a lot of work, but dammit. This utility is used by everyone in the RPM world.

**Comment by otheus** [comment]
big thumbs up (from me --let's hope the guy with the keys agrees)

**Comment by kdudka** [comment]
Thank you for working on this!  I think this is an easy-to-implement alternative to pull request #173, which tried to solve a bigger problem but in the end failed because of some corner-cases that were nearly impossible to address in a backward-compatible way.

`ignoreduplicatefilematches` feels a bit long.  What about `ignoreduplicates` instead?

I would also like to cover this by a test-case.

**Comment by falk-werner** [comment]
> `ignoreduplicatefilematches` feels a bit long. What about `ignoreduplicates` instead?

I like `ignoreduplicates` very much.
 
> I would also like to cover this by a test-case.

I added basic test just yet. The test ensures that no error occurs when the option is specified - otherwise an error occurs. I'm not sure about adding a negative test - should I?

**Review by kdudka — CHANGES_REQUESTED** [review]

File: test/test-0107.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+. ./test-common.sh
+
+cleanup 2
```
[the remark this answers is not in the document]
Comment (falk-werner) [reply test/test-0107.sh]: You are right. This is just a copy-&-paste mistake.
It should be correct now.

File: test/test-0107.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+# ------------------------------- Test 107 ------------------------------------
+preptest test.log 107 1
+$RLR test-config.107 --force || exit 23
```
[the remark this answers is not in the document]
Comment (falk-werner) [reply test/test-0107.sh:10]: Test case is extended as recommended.

**Review by kdudka — APPROVED** [review]

**Comment by kdudka** [comment]
@cgzones What is your take on this?

**Review by cgzones — APPROVED** [review]
Seems reasonable.

**Comment by kdudka** [comment]
@falk-werner I have squashed the fixup commits and slightly tweaked the commit messages.  Could you please confirm that we can merge the current version as it is?

**Comment by falk-werner** [comment]
@kdudka I confirm, please merge it.
